src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/Mutect2FilteringEngine.java

```diff
@@ -189,9 +199,40 @@ public VariantContext applyFiltersAndAccumulateOutputStats(final VariantContext
             }
         }
 
+        List<String> siteFilters = new ArrayList<>();
+        List<Map<Allele, String>> newFilters =
+                errorProbabilities.getProbabilitiesByFilterAndAllele().entrySet().stream().map(
+                entry -> addFilterStrings(entry.getValue(), siteFilters, entry.getKey().filterName())).collect(Collectors.toList());
+
+        siteFilters.stream().forEach(vcb::filter);
+        List<String> orderedASFilterStrings = vc.getAlleles().stream().map(allele -> getMergedFilterString(allele, newFilters)).collect(Collectors.toList());
+        String finalAttrString = AnnotationUtils.encodeAnyASList(orderedASFilterStrings);
+
+        vcb.putAttributes(Collections.singletonMap(GATKVCFConstants.AS_FILTER_STATUS_KEY, finalAttrString));
         return vcb.make();
     }
 
+    private String getMergedFilterString(Allele allele, List<Map<Allele, String>> newFilters) {
+        Set<String> results = newFilters.stream().map(m -> m.get(allele)).collect(Collectors.toSet());
```

Comment:
Collectors.toSet creates unordered HashSets which can cause non-determinism errors.  Here you're going from a set -> list so seems like you probably care about the order.

Reply:
thanks. done for both comments.